dali/operators/image/crop/bbox_crop.cc

```diff
@@ -14,143 +14,655 @@
+      make_string("Argument `", name, "` must be a ", ndim, "D vector"));
+
+    TensorShape<> sh(std::vector<int64_t>(tmp.begin(), tmp.end()));
+    v.resize(batch_size, sh);
+  } else {
+    if (required)
+      DALI_FAIL(make_string("Argument `", name, "` is required"));
+  }
+}
+
+struct SampleOption {
+  bool no_crop = false;
+  float min_iou = 0.0f;
+  SampleOption(bool _no_crop, float _min_iou)
+      : no_crop(_no_crop), min_iou(_min_iou) {}
+};
+
+struct Range {
+  Range(float _min, float _max)
+      : min(_min), max(_max) {
+    DALI_ENFORCE(min >= 0.0f,
+      make_string("Min should be at least 0.0. Received: ", min));
+    DALI_ENFORCE(min <= max,
+      make_string("Range should be provided as: [min, max]. Received: [", min, max, "]"));
+  }
+
+  explicit Range(const std::vector<float>& range)
+      : Range(range.size() == 2 ? range[0] : -1.0f,
+              range.size() == 2 ? range[1] : -1.0f) {
+  }
+
+  bool Contains(float k) const {
+    return k >= min && k <= max;
+  }
+
+  const float min, max;
+};
+
+struct ProspectiveCrop {
+  bool success = false;
+  BoundingBox crop{};
+  std::vector<BoundingBox> boxes;
+  std::vector<int> labels;
+
+  ProspectiveCrop(bool success, const BoundingBox &crop, const std::vector<BoundingBox> &boxes,
+                  const std::vector<int> &labels)
+      : success(success), crop(crop), boxes(boxes), labels(labels) {}
+  ProspectiveCrop() = default;
+};
+
+}  // namespace
+
 DALI_SCHEMA(RandomBBoxCrop)
     .DocStr(
-        R"code(Perform a prospective crop to an image while keeping bounding boxes and labels consistent. Inputs must be supplied as
-two Tensors: `BBoxes` containing bounding boxes represented as `[l,t,r,b]` or `[x,y,w,h]`, and `Labels` containing the
-corresponding label for each bounding box. Resulting prospective crop is provided as two Tensors: `Begin` containing the starting
-coordinates for the `crop` in `(x,y)` format, and 'Size' containing the dimensions of the `crop` in `(w,h)` format.
-Bounding boxes are provided as a `(m*4)` Tensor, where each bounding box is represented as `[l,t,r,b]` or `[x,y,w,h]`. Resulting
-labels match the boxes that remain, after being discarded with respect to the minimum accepted intersection threshold.
-Be advised, when `allow_no_crop` is `false` and `thresholds` does not contain `0` it is good to increase `num_attempts` as otherwise
-it may loop for a very long time.)code")
-    .NumInput(2)
-    .NumOutput(4)
+        R"code(Performs a prospective random crop to an image coordinate space while keeping
+bounding boxes (and optionally labels) consistent. That is, after applying the random crop operator
+to the image coordinate space, the bounding boxes will be adjusted or filtered out to match the
+cropped region of interest. The applied random crop operator is constrained by the arguments
+provided to the operator.
+
+The cropping window random selection is selected at random until it matches the overlap restrictions
```

Comment:
I would mention that `at most num_attempts times`.

Reply:
Actually it can loop forever. The logic goes like this:
```
while not_valid(crop):
    threshold = pick(thresholds)
    while (not_valid(crop) and attemtps < max_attemtps)
       crop = generate_crop(threshold)
```

Reply:
Dose the reference do the same (or the paper allows that)?

Reply:
Not sure about the paper, but that is the behavior we had here and I did not change it

Reply:
Just asking